src/main/java/io/nats/client/impl/NatsKeyValue.java

```diff
@@ -35,32 +37,37 @@
     private final String bucketName;
     private final String streamName;
     private final String streamSubject;
-    private final String defaultKeyPrefix;
-    private final String publishKeyPrefix;
+    private final String rawKeyPrefix;
+    private final String pubSubKeyPrefix;
 
     NatsKeyValue(NatsConnection connection, String bucketName, KeyValueOptions kvo) throws IOException {
         this.bucketName = Validator.validateKvBucketNameRequired(bucketName);
         streamName = toStreamName(bucketName);
         streamSubject = toStreamSubject(bucketName);
-        defaultKeyPrefix = toKeyPrefix(bucketName);
+        rawKeyPrefix = toKeyPrefix(bucketName);
```

Comment:
Nitpick...  Could simplify a bit and remove rawKeyPrefix, just assign to pubSubKeyPrefix and reassign if necessary.

Reply:
rawKeyPrefix is the base. pubSubKeyPrefix varies depending on whether or not there is a JetStream prefix